### ISSUE TITLE
UX pillar information are not always secret

### ISSUE THREAD

**Issue by dawidmalina** [erwindon/SaltGUI#100]
Pillar information are not always secret so it would be good to define list of pillar values or maybe list of pillar regexp values which should be hidden. For example in my example pillar called `malina` is not confidential so i would like to have for it regular output. 
<img width="922" alt="zrzut ekranu 2018-12-26 o 21 00 37" src="https://user-images.githubusercontent.com/5910851/50456028-dca03a80-0951-11e9-88fc-7e0d1036bd31.png">

<img width="923" alt="zrzut ekranu 2018-12-26 o 21 03 15" src="https://user-images.githubusercontent.com/5910851/50456030-e2961b80-0951-11e9-867a-c5208cedba77.png">

Ideally output should be formatted as mentioned in https://github.com/erwindon/SaltGUI/issues/99

**Comment by erwindon** [comment]
@dawidmalina 
Interesting suggestion!
Of course SaltGUI does not know this in advance and needs some additional input.
We can use the `master` file for this additional information.
My proposal would be to add an optional configuration parameter there called `saltgui_public_pillars`. This parameter then contains a list of regular expressions. When this parameter is not found, or it is not a list, the list is ignored. Invalid regular expressions are individually ignored.
I will implement this in the near future...

**Comment by erwindon** [comment]
I've created a possible solution in branch `publicpillars`.
Instructions for configuration are in the README.MD file.
Can you give it a try?

**Comment by dawidmalina** [comment]
I've try but it didn't work or I've configured it wrong.

Settings I've added to master config:
```
saltgui_public_pillars:
    - malina.*
```

Pillars which I have:
```
6dc1c985c3e9:
    ----------
    malina:
        ----------
        list:
            - a
            - b
        map:
            ----------
            m:
                ap
    password:
        dq4wh7nRTNq
```

But pillar `malina` is still hidden
<img width="923" alt="zrzut ekranu 2018-12-27 o 16 14 18" src="https://user-images.githubusercontent.com/5910851/50484768-90f19d80-09f2-11e9-8f91-74e8530720b4.png">

**Comment by dawidmalina** [comment]
Now I was able to make it work for me. But two small changes are required:
- small bug in js (marked in the pull request)
- small permission still missing
```
      - '@wheel':
        - 'key.*'
        - 'config.values'
```

**Comment by erwindon** [comment]
Ah, bugger, I never updated my `master` file with the new defaults after the previous maintainer changed them.  Time for some cleanup.

**Comment by erwindon** [comment]
I changed the description for the permissions to properly describe the current situation.
The branch `publicpillars` was re-based to include that.
Can you please give it another test?

**Comment by erwindon** [comment]
merged, see #102
